Versioning.md

```diff
@@ -0,0 +1,76 @@
+# Versioning
+
+This document describes the versioning policy for this repository.
+
+## Goals
+
+### API Stability
+
+Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, that API module will function with any SDK that has the same major version, and equal or greater minor or patch version.
+
+For example, libraries that are instrumented with `opentelemetry 1.0.1` will
+function in applications using `opentelemetry 1.11.33` or `opentelemetry
+1.3.4`.
+
+### SDK Stability
+
+Public portions of the SDK (constructors, configuration, end-user interfaces)
+must remain backwards compatible. Internal types are allowed to break.
+
+## Policy
+
+* Releases will follow [SemVer](https://semver.org/).
+* Only single source package containing api and sdk for all singals would be released as part of new GitHub release.
+* New telemetry signals will be introduced behind experimental feature flag usng C macro define.
+  ```
+    #ifdef METRICS_PREVIEW
```

Comment:
It'd be best to clarify on the best practices and coding convention here, such as `HAVE_ABC` or alike.

Reply:
done. Please feel free to provide suggestion if it's still not complete.